### ISSUE TITLE
Section headers and add fields

### ISSUE THREAD

**Issue by olantrust** [nextcloud/contacts#305]
1) Currently, in the App there are no section headers. Once you've entered a phone number, it no longer says phone number, each line of an address could be a field in itself.
2) Currently, to add another phone number, a User confusingly has to go to the end of a contact to 'Add field'.
So it is very confusing, and bad usability - because it is not clear without having to read and figure out what each field entry means, or where on the screen to find a particular detail, without reading through it.

This is a real barrier, and also looks less than graceful.

In all other modern systems,
1) is dealt with by having section headers.
2) is dealt with the user simply clicking a + sign by the relevant section header (User wants to add a phone number, simply clicks + by the 'Phone' header, and an additional phone field is created, populated by the default label 'Mobile' (which the User can alter if required).

**Solution**

Each header should have whitespace between.
Each header should have a + sign to add extra fields of that type.

**The section headers should be:**
Tel
Email
Address 
(to include CardDAV 'Location (GEO)' and Timezone (TZ), Language (LANG) fields if synced or implemented)

**The following headers would appear if a relevant field was added:**
_Messaging_
_Social_
_Web _
_Dates _
(to include birthday, anniversary, deathdate, deathplace, Free/busy(FBURL) if implemented/synced, and any custom fields of 'date' type)]
_Notes_
_Images_
(to include additional photos (vCard does not restrict the number of pics) and Logo field)
_Add more.._
_Add advanced..._

**Comment by skjnldsv** [comment]
I tend to aim for simplicity, but you have my attention! :)
Could you propose a mockup of what you suggest?
I'm not really sure how to implement this without breaking the nice fit we have between all of our properties.

**Comment by olantrust** [comment]
We'll post a mockup. 👍

**Comment by j-ed** [comment]
Point **3.** is indeed a pain. You always have to scroll to the end of a record to add a new field. I would recommend to add a button to the top of the record where the download and delete buttons are shown.

**Comment by Henni** [comment]
@jancborchardt and I worked on something like this.
But I lost my work when my dev instance overwrote my git repository during an update... 😢 

Might be a good thing to work on during the hackathon.

**Comment by jancborchardt** [comment]
> But I lost my work when my dev instance overwrote my git repository during an update

What the f … can you open an issue about this in the server repository? That is a major issue which should not happen. I remember it came up at the Winzerhof-Hackathon @LukasReschke @ChristophWurst @rullzer

**Comment by ChristophWurst** [comment]
> But I lost my work when my dev instance overwrote my git repository during an update... 😢

Did it also happen to you when you switched to an app branch where the version number was behind the latest release? Because that's why I twice lost my Mail dev files as Nextcloud replaced the local app directory with a tarball from the app store.

**Comment by Henni** [comment]
@ChristophWurst yes, that is exactly what happened. I don't know how this could be solved by the server repo... Maybe offering to skip upgrade if debug is enabled? Or is there already a command line option for the upgrade script which prevents this?

Also keeping the `.git` directory would have sufficed in my case.

**Comment by ChristophWurst** [comment]
It's simple: if an app is downgraded, then don't replace it. Alternatively, don't run the *update everything that can be updated* after an app update/downgrade.

**Comment by jancborchardt** [comment]
@ChristophWurst @Henni yep, or is it possible to check if the app is checked out via git? In that case it should of course not be upgraded with one from the app store.

**Comment by jancborchardt** [comment]
Opened a separate issue about the core problem at https://github.com/nextcloud/server/issues/7011

Let’s keep this issue on topic for working on the sections. :)
